src/pages/ListPage.tsx

```diff
@@ -1,28 +1,81 @@
 import { useHistory } from 'react-router-dom'
-import { Character, CharacterGender, CharacterStatus } from '../api/types'
+import { Character } from '../api/types'
 import { CharacterCard } from '../components/CharacterCard'
+import { FilterCard } from 'components/FilterCard'
+import { useHttp } from 'hooks/http.hook'
+import { useQuery } from 'react-query'
+import { useAppSelector } from 'hooks/useAppSelector'
+import { MouseEvent } from 'react'
+import { Alert, AlertTitle, Box, Button, Skeleton } from '@mui/material'
+import { useTypeDispatch } from 'hooks/useTypeDispatch'
+import { changeStatusActiveFilter, changeGenderActiveFilter } from 'store/reducers/filter'
+
+type requestType = {
+  info: {
+    count: number
+    pages: number
+    next: string
+    prev: string
+  }
+  results: Character[]
+}
 
 export const ListPage = () => {
+  const { activeStatus, activeGender, status, gender } = useAppSelector(state => state.filter)
+  const dispatch = useTypeDispatch()
+  const { request } = useHttp()
+
+  const fetchCharacters = async () => {
+    let query

```

Comment:
preferujeme vždy const, vede to i k lepším návykům, například vyhodit do samostatné metody buildQueryParams přepsané na:

const buildQueryParams = () => {
    if (activeStatus === 'All' && activeGender === 'All') {
      return ''
    }
    
    if (activeStatus === 'All' && activeGender !== 'All') {
      return `?gender=${activeGender}`
    }
    
    if (activeStatus !== 'All' && activeGender === 'All') {
      return `?status=${activeStatus}`
    }
    
    return `?&status=${activeStatus}&gender=${activeGender}`
}